.github/workflows/python-main.yml

```
@@ -384,14 +401,17 @@ jobs:
 
   combine-macos-wheels:
-    needs: build-macos
+    needs: [precheck, build-macos]
     strategy:
       matrix:
-        os: [macos-15-intel, macos-14]
+        os: ${{ fromJson(needs.precheck.outputs.wheel_macos_runners) }}
       fail-fast: false
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v3
```

actionlint (1.7.12)

[error] the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)